src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -93,12 +111,29 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
     }
 
     async deinitialize(): Promise<InternalCheckoutSelectors> {
+        clearInterval(this._pollingInterval);
         this._orderId = undefined;
         this._paypalCommercePaymentProcessor.deinitialize();
 
         return Promise.resolve(this._store.getState());
     }
 
+    private setPollingMechanism(gatewayId: string | undefined, submitForm: any) {
+        this._pollingInterval = setInterval(async () =>  {
+            try {
+                if (gatewayId === PaymentStrategyType.PAYPAL_COMMERCE_ALTERNATIVE_METHODS) {
+                    const res = await this._paypalCommerceRequestSender.getOrderStatus();
+                    if (res.status.toLowerCase() === APPROVED) {
```

Comment:
what happens if the order is never approved? How long will this keep polling for? Is there any chance that a process got stuck in a "pre-approved" status? We have other payment methods where we redirect to the order confirmation page despite the order not being approved but "placed". We display a message to the shopper notifying that the order is still pending manual verification from the merchant. Maybe this is something you would want to explore?